src/sys/wait.rs

```diff
@@ -64,74 +140,104 @@ libc_bitflags!(
         #[cfg(any(target_os = "android", target_os = "linux", target_os = "redox"))]
         #[cfg_attr(docsrs, doc(cfg(all())))]
         __WNOTHREAD;
-        /// Wait on all children, regardless of type
+        /// Wait for all children, regardless of type (clone or non-clone)
         #[cfg(any(target_os = "android", target_os = "linux", target_os = "redox"))]
         #[cfg_attr(docsrs, doc(cfg(all())))]
         __WALL;
-        /// Wait for "clone" children only.
+        /// Wait for "clone" children only. If omitted then wait for "non-clone" children only.
+        /// (A "clone" child is one which delivers no signal, or a signal other than [`SIGCHLD`] to
+        /// its parent upon termination.) This option is ignored if [`__WALL`] is also specified.
+        ///
+        /// [`SIGCHLD`]: crate::sys::signal::Signal::SIGCHLD
+        /// [`__WALL`]: Self::__WALL
         #[cfg(any(target_os = "android", target_os = "linux", target_os = "redox"))]
         #[cfg_attr(docsrs, doc(cfg(all())))]
         __WCLONE;
     }
 );
 
-/// Possible return values from `wait()` or `waitpid()`.
+/// Possible return values from [`wait`] or [`waitpid`].
 ///
-/// Each status (other than `StillAlive`) describes a state transition
-/// in a child process `Pid`, such as the process exiting or stopping,
+/// Each status (other than [`StillAlive`](WaitStatus::StillAlive))
+/// describes a state transition in a child process
+/// [`Pid`](crate::unistd::Pid), such as the process exiting or stopping,
 /// plus additional data about the transition if any.
 ///
-/// Note that there are two Linux-specific enum variants, `PtraceEvent`
-/// and `PtraceSyscall`. Portable code should avoid exhaustively
-/// matching on `WaitStatus`.
+/// Note that there are two Linux-specific enum variants,
+/// [`PtraceEvent`](WaitStatus) and [`PtraceSyscall`](WaitStatus).
```

Comment:
Are these links correct or did you mean to link to the specific enum variants?

Reply:
Listing the variants breaks CI builds on the platforms where those variants don't exist. This was my solution after I couldn't figure out how to make cfg apply to the comment block instead of the enum below. Happy to make the change if you know how.